src/main/kotlin/com/exactpro/th2/codec/hand/CodecHandFactory.kt

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2024 Exactpro (Exactpro Systems Limited)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.exactpro.th2.codec.hand
+
+import com.exactpro.th2.codec.api.IPipelineCodec
+import com.exactpro.th2.codec.api.IPipelineCodecContext
+import com.exactpro.th2.codec.api.IPipelineCodecFactory
+import com.exactpro.th2.codec.api.IPipelineCodecSettings
+import com.google.auto.service.AutoService
+import java.io.InputStream
+
+@AutoService(IPipelineCodecFactory::class)
+class CodecHandFactory : IPipelineCodecFactory {
+    override val settingsClass: Class<out CodecHandSettings>
+        get() = CodecHandSettings::class.java
+
+    override val protocols: Set<String>
+        get() = PROTOCOLS
+
+    override fun create(settings: IPipelineCodecSettings?): IPipelineCodec {
+        if (settings is CodecHandSettings) {
+            return CodecHand(settings)
+        }
+        throw IllegalArgumentException("Unexpected setting type: " + (settings?.javaClass ?: "null"))
```

Comment:
This block can be replaced with something like this. This is the same code but a bit more "Kotlinish" in my opinion. Feel free to keep the original code, I don't insist
```suggestion
        require(settings is CodecHandSettings) { "Unexpected setting type: ${settings?.javaClass}" }
        return CodecHand(settings)
```